modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/SourceAdapter.java

```diff
@@ -199,7 +216,19 @@ public DefaultSourceCompletionHandler(SourceCallbackExecutor onSuccessExecutor,
 
     @Override
     public Publisher<Void> onFailure(MessagingException exception, Map<String, Object> parameters) {
-      return from(onErrorExecutor.execute(exception.getEvent(), parameters, context))
+      final CoreEvent event = exception.getEvent();
```

Comment:
This needs to change in a future PR right? ModuleFlowProcessingPhase shouldn't really even call onFailure on back-pressue should it, or are you ok with keeping this here?

Reply:
That's actually a question for you. I did it here because this is where you are communicating the error. If you start sending it somewhere else, fine by me